homeassistant/components/mqtt/__init__.py

```diff
@@ -597,6 +597,14 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
             )
         )
 
+    if not bool(hass.config_entries.async_entries(DOMAIN)):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_DISCOVERY},
```

Comment:
`SOURCE_DISCOVERY` is for the `discovery` integration, use `SOURCE_INTEGRATION_DISCOVERY` instead

```suggestion
                context={"source": config_entries.SOURCE_INTEGRATION_DISCOVERY},
```

Although I do think we don't have a default handler for that... 🤔

Reply:
A default handler is added in [98ed5cf](https://github.com/home-assistant/core/pull/66248/commits/98ed5cfc705ad5bcd751d7ddde4715489c2799ca)